src/main/kotlin/no/nav/familie/ks/sak/barnehagelister/BarnehagelisteVarslingService.kt

```diff
@@ -54,24 +53,58 @@ class BarnehagelisteVarslingService(
     }
 
     // Bruker kommuneNr i stedet for kommunenavn siden det er mindre mulighet for feilskriving
-    private fun finnKommunerSendtInnSisteDøgn(): Set<String> {
+    private fun finnKommunerSendtInnSisteDøgn(): Set<KommuneEllerBydel> {
         val barnSendtInnSisteDøgn =
             barnehageBarnRepository
                 .findAll()
                 .filter { it.endretTidspunkt >= LocalDateTime.now().minusDays(1) }
-        return barnSendtInnSisteDøgn.map { it.kommuneNr }.toSet()
+        return barnSendtInnSisteDøgn.map { KommuneEllerBydel(it.kommuneNr, it.kommuneNavn) }.toSet()
+    }
+
+    private fun hentEnhetTilFylkeMap(): Map<String, Set<String>> {
+        val enhetTilFylkeFil = Thread.currentThread().contextClassLoader.getResource(PATH_ENHETSNR_TIL_FYLKENR_OVERSIKT)
+
+        return objectMapper.readValue(enhetTilFylkeFil, object : TypeReference<Map<String, Set<String>>>() {})
+    }
+
+    private fun hentAnsvarligEnhetForKommuneEllerBydelMap(enhetTilFylkeMap: Map<String, Set<String>>): Map<String, Set<KommuneEllerBydel>> =
+        enhetTilFylkeMap.mapValues { (_, fylkeKoder) ->
+            buildSet {
+                fylkeKoder.forEach { fk ->
+                    val kodeTilNavnIFylke = geografiService.hentBydelEllerKommuneKodeTilNavnFraFylkeNr(fk)
+                    kodeTilNavnIFylke.forEach { (kode, navn) ->
+                        add(KommuneEllerBydel(kode = kode, navn = navn))
+                    }
+                }
+            }
+        }
+
+    private fun finnEnhetForKommuneEllerBydel(
+        kode: String,
+        enhetTilFylkeMap: Map<String, Set<String>>,
+    ): String {
+        val ansvarligMap = hentAnsvarligEnhetForKommuneEllerBydelMap(enhetTilFylkeMap)
+        val kodeTilEnhet = ansvarligMap.flatMap { (enhet, set) -> set.map { it.kode to enhet } }.toMap()
+
+        return kodeTilEnhet[kode]
+            ?: throw Feil("Ingen enheter har ansvar for kommunen/bydelen $kode")
     }
 
     companion object {
         val `KONTAKT_E-POST_VADSØ` = "nav.familie-.og.pensjonsytelser.vadso.kontantstotte@nav.no"
         val `KONTAKT_E-POST_BERGEN` = "kontantstotte@nav.no"
-        val PATH_ENHETSNR_TIL_KOMMUNENR_OVERSIKT =
-            "barnehagelister/barnehagelister-enhet-til-kommune.json"
+        val PATH_ENHETSNR_TIL_FYLKENR_OVERSIKT =
+            "barnehagelister/barnehagelister-enhet-til-fylkeNr.json"
 
         private val logger = LoggerFactory.getLogger(BarnehagelisteVarslingService::class.java)
     }
 }
 
+data class KommuneEllerBydel(
+    val kode: String,
+    val navn: String,
+)
```

Comment:
Det hadde vært litt lettere å forstå andre steder i koden hvis `kode` het `nummer` i stedet for, så man skjønner at det er kommunenummer eller bydelsnummer det er snakk om